### PR TITLE
Publish Allure reports to GitHub pages

### PR DESCRIPTION
**Pull request by szczys**
add a workflow to gather all individual reports, publish them as an artifact, generate the Allure webpage, and save it to the `gh-pages` branch for automatic publishing to GitHub pages.

- [ ] Set up automatic deploy of `gh-pages` branch to GitHub Pages

resolves https://github.com/golioth/firmware-issue-tracker/issues/663

### CODE REVIEW

File: .github/workflows/allure-publish-reports.yml
Comment (szczys) [inline .github/workflows/allure-publish-reports.yml:7]:
```suggestion
  allure-collect-and-publish:
```

File: .github/workflows/allure-publish-reports.yml
Comment (szczys) [inline .github/workflows/allure-publish-reports.yml:1]: The name of this file should be changed to `reports-allure-publish.yml` so that it is grouped with other report-related workflows (there's one for `reports-summary-publish.yml` coming soon).

**Comment by szczys** [comment]
Okay, this one's deploying [over to the other repo](https://github.com/golioth/allure-reports/tree/gh-pages) now and ready for review.

I haven't turned on the public page publishing yet.

**Review by mniestroj — APPROVED** [review]
Nothing blocking here, so would like to see the results it generates / publishes.

**Comment by sam-golioth** [comment]
Let's turn on the publishing so we can make sure the whole thing works. There are reference to this repo in that one that I think could cause problems (e.g. https://github.com/golioth/allure-reports/blob/gh-pages/index.html#L1)

**Comment by szczys** [comment]
I have done the following:
- Removed history that had secrets in it. All future pushes will use the safe-upload-artifacts to mask secrets.
- Fixtured out the undocumented inputs for specifying the site URL mentioned by @sam-golioth above
- Enabled auto publishing of github pages. The next time this pushes the reports should appear here: https://golioth.github.io/allure-reports/

I am waiting until the workflow cancellation bug fix is merged to main before pushing the updated version of this PR (which will kick off a new Allure build).

File: .github/workflows/reports-allure-publish.yml
Comment (sam-golioth) [inline .github/workflows/reports-allure-publish.yml:16]: Do we need this? There's nothing before it.

**Review by sam-golioth — APPROVED** [review]
Wooo! Excited to have this! Just a few small comments :)

**Pull request merged**
